src/com/sap/piper/cm/ChangeManagement.groovy

```diff
@@ -114,37 +114,87 @@ public class ChangeManagement implements Serializable {
-        int rc = executeWithCredentials( endpoint, credentialsId, 'release-transport', ['-cID', changeId,
-                                                                                        '-tID', transportRequestId], clientOpts) as int
+        int rc = executeWithCredentials( endpoint, credentialsId, 'release-transport', [new KeyValue('cID', changeId),
+                                                                                        new KeyValue('tID', transportRequestId)], clientOpts) as int
         if(rc == 0) {
             return
         } else {
             throw new ChangeManagementException("Cannot release Transport Request '$transportRequestId'. Return code from cmclient: $rc.")
         }
     }
-
+    
     String getCMCommandLine(String endpoint,
                             String username,
                             String password,
                             String command,
-                            List<String> args,
+                            List<CLOption> args,
                             String clientOpts = '') {
         String cmCommandLine = '#!/bin/bash'
         if(clientOpts) {
             cmCommandLine +=  """
                              export CMCLIENT_OPTS="${clientOpts}" """
         }
+        
         cmCommandLine += """
                         cmclient -e '$endpoint' \
                            -u '$username' \
                            -p '$password' \
                            -t SOLMAN \
-                          ${command} ${(args as Iterable).join(' ')}
+                          ${command} ${args.collect{it}.join(' ')}
                     """
+        
         return cmCommandLine
     }
+    
+    abstract static class CLOption 
+    {
+        String key=null
+        String value=null
+        boolean quoteValue=true;
+        String hyphen='-'
+        
+        protected CLOption(String key, String value) {
+            this.key = key
+            this.value = value
+        }
+        CLOption setQuotes(boolean quoteValue) {
+            this.quoteValue=quoteValue
+            return this
+        }
+        CLOption setHyphen(String hyphen) {
+            this.hyphen=hyphen
+            return this
+        }
+        String toString() {
+            StringBuilder  sb=new StringBuilder()
+            if(key!=null) {
+                sb.append("${hyphen}${key}")
+            }
+            if(value!=null) {
+                if(sb.length()!=0) {
+                    sb.append(" ")
+                }
+                if(quoteValue) {
+                    sb.append("'$value'")
+                }
+                else {
+                    sb.append(value)
+                }
+            }
+            return sb.toString()
+        }
+    }
+    
+    public static final class KeyValue extends CLOption     {
+        KeyValue(String key, String value) {super(key?:'null',value ?: 'null')}
```

Comment:
Why handing over the string 'null' in case there is no key (or value)?

Reply:
removed. exception will be thrown now